lib/compat/wordpress-6.7/block-templates.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Block template functions.
+ *
+ * @package gutenberg
+ */
+
+if ( ! function_exists( 'wp_register_template' ) ) {
+	/**
+	 * Register a template.
+	 *
+	 * @param string       $template_name  Template name in the form of `plugin_uri//template_name`.
+	 * @param array|string $args           Object type or array of object types with which the taxonomy should be associated.
+	 * @param array|string $args           {
+	 *     @type string        $title                 Optional. Title of the template as it will be shown in the Site Editor
+	 *                                                and other UI elements.
+	 *     @type string        $description           Optional. Description of the template as it will be shown in the Site
+	 *                                                Editor.
+	 *     @type string        $content               Optional. Default content of the template that will be used when the
+	 *                                                template is rendered or edited in the editor.
+	 *     @type string[]      $post_types            Optional. Array of post types to which the template should be available.
+	 *     @type string        $plugin                Uri of the plugin that registers the template.
+	 * }
+	 * @return WP_Block_Template|WP_Error The registered template object on success, WP_Error object on failure.
+	 */
+	function wp_register_template( $template_name, $args = array() ) {
```

Comment:
Small terminology question:

We have functions like `get_block_template()` should these be `wp_register_block_template()` instead?

Reply:
I made the change after [this comment](https://github.com/WordPress/gutenberg/pull/61577#discussion_r1648831053), but after reading yours I'm unsure, I get your point about the misalignment with `get_block_template()`.  :thinking:  Should we go back to `wp_register_block_template()` and `WP_Block_Templates_Registry`?

Reply:
I don't have a strong opinion on it — cc @youknowriad @gziolo @luisherranz for thoughts

Reply:
@SantosGuillamot should have good insights as we went through a similar process with Block Bindings and it wasn't easy to choose based on prior examples. We ended up with:
- WP_Block_Bindings_Source
- WP_Block_Bindings_Registry
- register_block_bindings_source
- get_block_bindings_source

There was also a nuance raised after the fact where in spoken language you say gets block binding source, but we used consistently plural form "bindings" in all singatures.

I don't have strong opinions here. Mostly sharing the latest work with similar considerations.
It would be useful to document it somewhere to make the decision process easier for future contributions 😀

In particular, with the developer docs it should not matter that much as you can search using keywords or even arrive on the correct signature through Google.

Reply:
Thanks for all the feedback! In f73a40480e04e3d5bc51b1112b8d2fab57558a38 I changed the following:

* `wp_register_template` → `wp_register_block_template`
* `wp_unregister_template` → `wp_unregister_block_template`
* `WP_Templates_Registry` → `WP_Block_Templates_Registry`
* and some internal texts and strings used in tests where it made sense

Reply:
In bindings, we used the `block` prefix to keep consistency with other APIs like `register_block_type`, `register_block_pattern`, `registerBlockVariation`, `registerBlockCollection`...

However, those are clearly related to blocks. In the case of `templates` and `template_parts`, I seem them more related to themes and not blocks.

I can see we have:

* [get_template](https://developer.wordpress.org/reference/functions/get_template/): Returns the current active theme.
* [get_block_template](https://developer.wordpress.org/reference/functions/get_block_template/): Returns the template.
* [get_template_part](https://developer.wordpress.org/reference/functions/get_template_part/): Returns the template part.

It feels a bit weird to me, so I am not sure what is the best approach.